### PR TITLE
Backend: Endpoint products sorting

### LINKED ISSUE #152: Backend: Endpoint for products improved with sortKey and sortOrder
## User story
**Who:** **As a** developer
**What:** **I want to** to be able to sort the products by name and price
**Why:** **so that I can** use them in fetching the products and sort them by calling the api.
## Acceptance criteria
1. Backend should fallback to default values if a parameter is not set in the URL.
2. Backend should throw 400 if type/value of any of the parameters is incorrect
3. Backend should make use of query parameters to sort the products correctly using order by and asc/desc.
4. Backend should return an empty list if the database does not return anything, e.g. offset is set to 10000

## Implementation Details
Products end point should suport two extra parameter sortKey and sortOrder fx: api/products?sortKey=price?sortOrder=desc
- sortKey, should have a default value of name, another possible value is price
- sortOrder, should have a default value of asc, another possible value is desc

Read about Swagger here https://github.com/HackYourFuture-CPH/fp-class18/#-working-with-api-documentation---swagger and this article should also help you with that https://blog.logrocket.com/documenting-your-express-api-with-swagger/

### PR DESCRIPTION
**Pull request by FlorentinaPetica**
# Description

Improved endpoind api/products to sort and order the products by fe. name, price in asc/desc order

Fixes #152

# How to test?

Pull the Branch, 
Run: npm run dev ,
Check endpoint like [http://localhost:3000/api/products?limit=5&offset=3&price=desc OR
go to: [http://localhost:3000/api/documentation] to check swagger documentation for the endpoint,

# Checklist

- [x] I have performed a self-review of my own code
- [x] I have followed the name conventions for CSS Classnames and filenames, Components names and filenames, Style filenames, if you are in doubt check the project README.MD and here https://github.com/HackYourFuture-CPH/curriculum/blob/master/review/review-checklist.md
- [x] I have commented on my code, particularly in hard-to-understand areas, if your code was simple enough mark the box anyway
- [x] I have made corresponding changes to the documentation, if your code was simple enough mark the box anyway
- [x] This PR is ready to be merged and not breaking any other functionality

### CODE REVIEW

**Comment by Impressiv3** [comment]
image

**Review by Impressiv3 — APPROVED** [review]
LGTM
You may push your recent change with the shorter code if you want to.

**Comment by FlorentinaPetica** [comment]
> LGTM You may push your recent change with the shorter code if you want to.

I used your suggestions. Thank you.

**Comment by Impressiv3** [comment]
We need one more approval :D

**Review by JiuTak — APPROVED** [review]
image
👍

**Review by Divyajg — COMMENTED** [review]
Everything is working perfect, just for one thing its showing response code 200 instead of 400!

image

File: src/server/api/controllers/products.controller.js [hunk 1]
```diff
@@ -3,22 +3,32 @@ const HttpError = require('../lib/utils/http-error');
...
 
-  if (limit && offset) {
-    products = products.limit(limit).offset(offset);
-  }
+    offset = offset || 0;
 
-  if (isNaN(limit) || isNaN(offset)) {
-    throw new HttpError('Limit and offset should be a number', 400);
-  }
+    sortKey = sortKey || 'name' || 'price';
```
Comment (pankaj28843) [inline src/server/api/controllers/products.controller.js]: 'name' will win if sortKey is not truthy. 'name' is always truthy as it is non empty string which is hard-coded.

**Review by pankaj28843 — APPROVED** [review]

**Pull request merged**